server/api/react.ts

```diff
@@ -0,0 +1,82 @@
+import { db } from "../database";
+import { Request, Response } from "express";
+import { sessions } from "../types/Session";
+
+let firstPost: boolean = true;
+let postID: number;
+
+/**
+ * Reacts to a post.
+ * @param {Request} req Requires a JSON body which includes an 'id' and 'token' field.
+ * @param {Response} res Response to request (200 if success, 400 if post does not exist, and 500 if failed for other reason)
+ */
+export function react(req: Request, res: Response) {
+    if ("id" in req.body && "token" in req.body) {
+        // make sure we have a valid token (one that finds us a valid user)
+        if (sessions.has(req.body.token)) {
+            const username = sessions.get(req.body.token).username;
+
+            // first, lets make sure the post exists
+            db.get(`SELECT * FROM Posts WHERE ID=?`, [req.body.id], function (err: Error, postRow: any) {
+                // error running query
+                if (err) {
+                    console.log("[SQL] Error: " + err);
+                    res.status(500).send("Database error!");
+                    return;
+                }
+
+                // if no post is found
+                if (!postRow || !postRow.test_validpost) {
+                    console.log('[API] Edit request failed! (invalid post ID)');
+                    res.status(400).send("Invalid post ID!");
+                    return;
+                }
+
+                // then, check if we have already disliked this post (we want to remove the dislike if that is the case)
+                db.get(`SELECT * FROM PostDislikes WHERE ID=? AND Username=?`, [req.body.id, username], function (err: Error, dislikeRow: any) {
+                    if (err) {
+                        console.log("[SQL] Error: " + err);
+                        res.status(500).send("Database error!");
+                        return;
+                    }
+                    if (!dislikeRow || !dislikeRow.test_disliked) {
```

Comment:
Where are you getting this `test_disliked` value? If it doesn't exist, this will always fail